shenyu-plugin/shenyu-plugin-apache-dubbo/src/main/java/org/apache/shenyu/plugin/apache/dubbo/proxy/ApacheDubboProxyService.java

```diff
@@ -106,4 +92,14 @@ public ApacheDubboProxyService(final BodyParamResolveService bodyParamResolveSer
             return ret;
         })).onErrorMap(exception -> exception instanceof GenericException ? new ShenyuException(((GenericException) exception).getExceptionMessage()) : new ShenyuException(exception));
     }
+
+    private CompletableFuture<Object> invokeAsync(final GenericService genericService, final String method, final String[] parameterTypes, final Object[] args) throws GenericException {
+        //Compatible with asynchronous calls of lower Dubbo versions
+        genericService.$invoke(method, parameterTypes, args);
+        Object resultFromFuture = RpcContext.getContext().getFuture();
+        if (resultFromFuture instanceof CompletableFuture) {
+            return (CompletableFuture<Object>) resultFromFuture;
+        }
+        return CompletableFuture.completedFuture(resultFromFuture);
```

Comment:
I found the code in `org.apache.dubbo.rpc.service.GenericService`
```
Object object = genericService.$invoke(method, parameterTypes, args);
        if (object instanceof CompletableFuture) {
            return (CompletableFuture<Object>) object;
        }
        return CompletableFuture.completedFuture(object);
```

Reply:
Please see here [Dubbo async call](https://dubbo.apache.org/zh/docs/advanced/async-call/)